x-pack/plugins/security_solution/public/common/components/upselling_provider/upselling_provider.tsx

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { memo, useContext } from 'react';
+import type { UpsellingService } from '../../..';
+
+export const UpsellingProviderContext = React.createContext<UpsellingService | null>(null);
+
+export type UpsellingProviderProps = React.PropsWithChildren<{
+  upsellingService: UpsellingService;
+}>;
+
+export const UpsellingProvider = memo<UpsellingProviderProps>(({ upsellingService, children }) => {
```

Comment:
Is there any specific reason to pass this through provider? 
Do you think we could use existing hooks instead? https://github.com/elastic/kibana/blob/b6d94d788362aa8aeddf81b3a1651fa389f0e454/x-pack/plugins/security_solution/public/common/hooks/use_upselling.ts

Reply:
The provider context for upselling is needed in order be able to use the upselling service hook from components that are rendered in Fleet (our UI extensions). Without it, we would not be able to re-use our components around this serverless logic (inside of our policy form).

Reply:
I see, so these extensions are built in SecuritySolution but used outside of it?